articles/user-guide/using-qsharp/control-flow.md

```diff
@@ -0,0 +1,329 @@
+    X(target);
+} 
+```
+or
+```qsharp
+if (i == 1) {
+    X(target);
+} elif (i == 2) {
+    Y(target);
+} else {
+    Z(target);
+}
+```
+
+## For Loop
+
+The `for` statement supports iteration over an integer range or over an array.
+The statement consists of the keyword `for`, an open parenthesis `(`, followed by a symbol or symbol tuple, the keyword `in`, an expression of type `Range` or array, a close parenthesis `)`, and a statement block.
+
+The statement block (the body of the loop) is executed repeatedly, with the defined symbol(s) (the loop variable(s)) bound to each value in the range or array.
+Note that if the range expression evaluates to an empty range or array, the body will not be executed at all.
+The expression is fully evaluated before entering the loop, and will not change while the loop is executing.
+
+The loop variable is bound at each entrance to the loop body, and unbound at the end of the body.
+In particular, the loop variable is not bound after the for loop is completed.
+The binding of the declared symbol(s) is immutable and follows the same rules as other variable bindings. 
+
+For some examples, supposing `qubits` is a register of qubits (i.e. of type `Qubit[]`), 
+
+```qsharp
+// ...
+for (qubit in qubits) {  // iterate over each qubit value in the array qubits
+    H(qubit);
+}
+
+mutable results = new (Int, Results)[Length(qubits)];
+for (index in 0 .. Length(qubits) - 1) {  // iterates over the integers in the Range 0 .. (Length(qubits) - 1)
+    set results w/= index <- (index, M(qubits[index])); // measure each qubit, updates the tuple value in the results array that at index 
+}
+
+mutable accumulated = 0;
+for ((index, measured) in results) { // iterates over the tuple values in results
+    if (measured == One) {
+        set accumulated += 1 <<< index;
+    }
+}
+```
+Note that at the end we utilized the arithmetic-shift-left binary operator, `<<<`, details of which can be found at [Numeric Expressions](xref:microsoft.quantum.guide.expressions#numeric-expressions)
+
+
+## Repeat-Until-Success Loop
+
+The Q# language allows classical control flow to depend on the results of measuring qubits.
+This capability in turn enables implementing powerful probabilistic gadgets that can reduce the computational cost for implementing unitaries.
+As an example, it is easy to implement so-called *Repeat-Until-Success* (RUS) patterns in Q#.
+These RUS patterns are probabilistic programs that have an *expected* low cost in terms of elementary gates, but for which the true cost depends on an actual run and an actual interleaving of various possible branchings.
+
+To facilitate Repeat-Until-Success (RUS) patterns, Q# supports the constructs
+
+```qsharp
+repeat {
+    statementBlock1
+}
+until (expression)
+fixup {
+    statementBlock2
+}
+```
+
+where `statementBlock1` and `statementBlock2` are zero or more Q# statements, and `expression` is any valid expression that evaluates to a value of type `Bool`.
```

Comment:
I felt like if we just say "Q# statements" it could accidentally imply that you can only put a single statement. Rather than go more verbose with exhaustive explanation, I updated the examples to be more generic. About to push it with the newest commit, let me know what you think.